processor/patch.go

```diff
@@ -0,0 +1,70 @@
+package processor
+
+import (
+	"bytes"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/token"
+)
+
+type patch struct {
+	pos   int
+	stmts []ast.Stmt
+}
+
+func (p *Processor) patchFile(fset *token.FileSet, file *ast.File, patches ...patch) error {
+	src, err := formatNode(fset, file)
+	if err != nil {
+		return err
+	}
+
+	var buf bytes.Buffer
+
+	offset := -int(file.FileStart) + 1
+	for _, patch := range patches {
+		buf.Reset()
+
+		buf.WriteString("\n")
+		if err := format.Node(&buf, fset, patch.stmts); err != nil {
+			return err
+		}
+		buf.WriteString("\n")
+
+		pos := patch.pos + offset
+		src = append(src[:pos], append(buf.Bytes(), src[pos:]...)...)
+		offset += buf.Len()
+	}
+
+	p.updateFile(fset, file, src)
+
+	return nil
+}
+
+func formatNode(fset *token.FileSet, node any) ([]byte, error) {
+	var buf bytes.Buffer
+	if err := format.Node(&buf, fset, node); err != nil {
+		return nil, err
+	}
+
+	return buf.Bytes(), nil
+}
+
+func (p *Processor) updateFile(fset *token.FileSet, file *ast.File, newSrc []byte) error {
```

Comment:
no need for method receiver, right? + renaming what is happening
```suggestion
func replaceFileWithSrc(fset *token.FileSet, file *ast.File, newSrc []byte) error {
```

Reply:
Also `processor.functionName`, `processor.isContext`, etc.. methods do not need a receiver.

I think we’ll leave it as it is, in the future we’ll probably store fset inside the processor.